osu.Server.Spectator.Tests/Multiplayer/MultiplayerTest.cs

```diff
@@ -35,26 +35,57 @@ public abstract class MultiplayerTest
         protected EntityStore<MultiplayerClientState> UserStates { get; }
 
         protected readonly Mock<IDatabaseFactory> DatabaseFactory;
-
         protected readonly Mock<IDatabaseAccess> Database;
 
-        protected readonly Mock<IMultiplayerClient> Receiver;
-        protected readonly Mock<IMultiplayerClient> GameplayReceiver;
+        /// <summary>
+        /// A general non-gameplay receiver for the room with ID <see cref="ROOM_ID"/>.
+        /// </summary>
+        protected readonly Mock<DelegatingMultiplayerClient> Receiver;
```

Comment:
This is ballooning a bit now in size. I think I'd prefer to keep this more compact/potentially flexible via some dictionary usage:

```csharp
protected readonly Dictionary<int, DelegatingMultiplayerClient> RoomGeneralReceivers;
protected readonly Dictionary<int, DelegatingMultiplayerClient> RoomGameplayReceivers;
protected readonly Dictionary<int, IMultiplayerClient> UserReceivers;
```

Unsure if out of scope to change this here. Also tempted to fold `ContextUser`s in a similar way but that is definitely out of scope here.

Reply:
Not going to do that in this PR because this isn't a refactoring PR.